src/API/Management/ResourceServers.php

```diff
@@ -53,6 +71,11 @@ public function update($id, $data)
     {
         $response = $this->httpClient->patch(sprintf('/resource-servers/%s', $id), [], json_encode($data));
 
-        return ResponseMediator::getContent($response);
+        // TODO Is this really correct?
+        if (201 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
```

Comment:
I've raised this one to be documented, it might get removed, if it does we can remove from API in another PR.